rig/machine_control/scp_connection.py

```
@@ -219,19 +381,31 @@ def write(self, buffer_size, x, y, p, address, data):
         # While there is still data perform a write: get the block to write
         # this time around, determine the data type, perform the write and
         # increment the address
-        end = len(data)
-        pos = 0
-        while pos < end:
-            block = data[pos:pos + buffer_size]
-            block_size = len(block)
+        def packets(address, data):
+            end = len(data)
+            pos = 0
+            while pos < end:
+                block = data[pos:pos + buffer_size]
+                block_size = len(block)
+
+                dtype = consts.address_length_dtype[(address % 4,
+                                                     block_size % 4)]
+
+                yield scpcall(x, y, p, consts.SCPCommands.write, address,
+                              block_size, dtype, block)
+
+                address += block_size
+                pos += block_size
 
-            dtype = consts.address_length_dtype[(address % 4, block_size % 4)]
+        # Run the event loop and then return the retrieved data
+        self.send_scp_burst(buffer_size, window_size, packets(address, data))
 
-            self.send_scp(buffer_size, x, y, p, consts.SCPCommands.write,
-                          address, block_size, dtype, block)
 
-            address += block_size
-            pos += block_size
+def seqs(mask=0xf):
```

Default mask can, I think, be safely `0xffff` (and should be).